src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/unit/clients/IGDBClientTests.kt

```diff
@@ -0,0 +1,177 @@
+package com.gamingbacklog.api.gamingbacklogapi.unit.clients
+
+import com.gamingbacklog.api.gamingbacklogapi.clients.IGDBClient
+import com.gamingbacklog.api.gamingbacklogapi.clients.util.ExternalAPIClient
+import com.gamingbacklog.api.gamingbacklogapi.unit.testutil.models.MockResponse
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.given
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.TestPropertySource
+
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+
+@SpringBootTest
+@Import(IGDBClient::class)
+@TestPropertySource(properties = ["CLIENT_ID=test_id", "CLIENT_SECRET=test_secret"])
+class IGDBClientTests {
+  lateinit var externalAPIClient: ExternalAPIClient
+  lateinit var igdbClient: IGDBClient
+
+  @BeforeEach
+  fun configureSystem() {
+    externalAPIClient = mock(ExternalAPIClient::class.java)
+    igdbClient = IGDBClient(externalAPIClient)
+  }
+
+  @Nested
+  @DisplayName("Tests for Authenticate")
+  inner class Authenticate {
+
+    @Test
+    fun shouldSuccessfullyReturnToken() {
+      given(externalAPIClient.postExternalCall(any(), eq(null), eq(null))).willReturn(
+        MockResponse(
+          "{\"access_token\":\"test_secret\",\"expires_in\":5180593,\"token_type\":\"bearer\"}",
+          200
+      )
+      )
+      val result = igdbClient.authenticate()
+      Assertions.assertEquals("test_secret", result.access_token)
+      Assertions.assertEquals("5180593", result.expires_in)
+      Assertions.assertEquals("bearer", result.token_type)
+    }
+
+    @Test
+    fun shouldErrorOnAuthentication() {
+      given(externalAPIClient.postExternalCall(any(), eq(null), eq(null))).willReturn(
+        MockResponse(
+          "{\"status\":400,\"message\":\"invalid\"}",
+          400
+        )
+      )
+      val result = igdbClient.authenticate()
+      Assertions.assertEquals("", result.access_token)
+      Assertions.assertEquals("", result.expires_in)
+      Assertions.assertEquals("", result.token_type)
+    }
+  }
+
+  @Nested
+  @DisplayName("Tests for Games Request")
+  inner class GamesRequest {
+    // successful return of game
+    @Test
+    fun shouldSuccessfullyReturnGame() {
+      given(externalAPIClient.postExternalCall(any(), any(), any())).willReturn(
+        MockResponse(
+          "[\n" +
```

Comment:
Repeat comment but might have been missed:
Do we want to define a class for the expected returns rather than just leave it as a string? (Can do in a separate ticket)

Reply:
Also repeated here for convenience:

It would be great if we could define a Kotlin class for these API responses to avoid these verbose strings in the future. We'd have to add in (de)serialization logic as well for that

Alternatively as a simpler short term approach, we could define a constants file for common use test strings to clean this up

Reply:
Created ticket, feel free to rephrase the description: https://gaming-backlog.atlassian.net/browse/GB-54 we'll definitely need this

for now, I refactored into a constants file, which isn't perfect but is a good short term solution like you said